### PR TITLE
Move chat command handling code from c++ to lua.

### PR DESCRIPTION
**Pull request by red-001**

### CODE REVIEW

**Comment by sofar** [comment]
Typo in title.

**Comment by sofar** [comment]
didn't test, but looks ok. :+1:

**Comment by red-001** [comment]
Fixed the typo

**Comment by nerzhul** [comment]
It's bit more slower than C++ but command handling is not performance critical

File: builtin/game/chatcommands.lua [hunk 1]
```diff
@@ -7,13 +7,23 @@
...
+	end
+
 	local cmd, param = string.match(message, "^/([^ ]+) *(.*)")
+	if not cmd then
+		core.chat_send_player(name, "-!- Empty command")
+		return true
+	end
 	if not param then
 		param = ""
 	end
```
Comment (sfan5) [inline builtin/game/chatcommands.lua]: replace this with `param = param or ""` since you're already editing here

Comment (red-001) [inline builtin/game/chatcommands.lua]: done

**Pull request merged**